services/common/src/components/documents/spatial/AddSpatialDocumentsModal.tsx

```diff
@@ -101,12 +101,29 @@ const AddSpatialDocumentsModal: FC<AddSpatialDocumentsModalProps> = ({
     },
   ];
 
+  const addBundleIdToFiles = ({
+    docman_bundle_guid,
+    geomark_id,
+  }: {
+    docman_bundle_guid: string;
+    geomark_id: string;
+  }) => {
+    const newFiles = currentValues[fieldName].map((f) => {
+      return {
+        ...f,
+        docman_bundle_guid,
+        geomark_id,
+        bundle_id: docman_bundle_guid,
+      };
+    });
+    dispatch(change(modalFormName, fieldName, [...initialDocuments, ...newFiles]));
+    dispatch(change(formName, fieldName, [...initialDocuments, ...newFiles]));
```

Comment:
ooooo one little maybe hiccup here-
what happens if the user uploads files and then cancels? That was the reason for only updating the parent form when `isFinalStep`.

Reply:
Hmmm tricky.  

I see what you're saying.  Since it's these documents that get sent to the viewSpatial component they needed to contain these new values.  Maybe can pass the other set of documents.